src/Microsoft.AspNetCore.Server.Kestrel/Internal/Http/KestrelHttpParser.cs

```diff
@@ -66,6 +68,27 @@ public KestrelHttpParser(IKestrelTrace log)
                 span = startLineBuffer.ToSpan();
             }
 
+            // pathStart
+            // |    pathEnd and queryStart
```

Comment:
do we need pathEnd _and_ queryStart? I couldn't think of a scenario when they are different.